homeassistant/components/sms/gateway.py

```diff
@@ -167,8 +167,13 @@ async def create_sms_gateway(config, hass):
     """Create the sms gateway."""
     try:
         gateway = Gateway(config, hass)
-        await gateway.init_async()
-        return gateway
+        try:
+            await gateway.init_async()
+            return gateway
```

Comment:
Please move this line down out of the try... except blocks.

Reply:
Done